master/master.cfg

```diff
@@ -46,32 +46,52 @@ WORKER_SECRET = Path('halide_bb_pass.txt').read_text().strip()
 WEBHOOK_SECRET = Path('webhook_token.txt').read_text().strip()
 WWW_PASSWORD = Path('buildbot_www_pass.txt').read_text().strip()
 
+# SERVER SETTINGS
+
+ARTIFACTS_DIR = os.environ.get('HALIDE_BB_MASTER_ARTIFACTS_DIR', '/home/abadams/artifacts')
+REPO_DIR = Path(__file__, '..', '..').resolve()
+
 # LLVM
 
 # At any given time, we test (at least) 3 LLVM versions:
 # - the current trunk (changes daily)
 # - the most recent release (expected to be stable)
 # - an older release (expected to be stable)
 #
-# the branches that correspond to these will rotate as new versions
-# are released, but the underlying test logic should not need changing
+# The branches that correspond to these will rotate as new versions
+# are released, but the underlying test logic should not need changing.
 
-LLVM_TRUNK_BRANCH = 'main'
-LLVM_RELEASE_BRANCH = 'release/11.x'
-LLVM_OLD_BRANCH = 'release/10.x'
+Version = namedtuple('Version', ['major', 'minor', 'patch'])
+VersionedBranch = namedtuple('VersionedBranch', ['ref', 'version'])
 
-_LLVM_BRANCHES = [LLVM_TRUNK_BRANCH, LLVM_RELEASE_BRANCH, LLVM_OLD_BRANCH]
+LLVM_TRUNK = 'trunk'
+LLVM_RELEASE_11 = 'release_11'
+LLVM_RELEASE_10 = 'release_10'
 
-_TO_VERSION = {
-    LLVM_TRUNK_BRANCH: 12,
-    LLVM_RELEASE_BRANCH: 11,
-    LLVM_OLD_BRANCH: 10,
-}
+LLVM = {LLVM_TRUNK: VersionedBranch(ref='main', version=Version(12, 0, 0)),
```

Comment:
I feel vague misgivings about `LLVM` and `HALIDE` being the identifiers here; that's awfully generic and feels like it's missing something. I'd vote for adding a suffix like _BRANCHES or _VERSIONS or something.

Reply:
I can add `_BRANCHES` as a suffix if you prefer.